### ISSUE TITLE
Slurm环境下硬件监控无数据

### ISSUE THREAD

**Issue by LiaoLW** [SwanHubX/SwanLab#1365]
### 确认项

- [x] 我已确认我遇到的问题在[文档](https://docs.swanlab.cn)中未有说明。
- [x] 我已在 [GitHub](https://github.com/SwanHubX/SwanLab/issues) 上搜索过类似或已有的问题。
- [x] 我已确认此问题并非由网络错误引起。

### 问题描述

在实验记录过程中，有自主log的训练数据，但是在系统标签页下没有硬件监控数据，显示为暂无图表

<img width="2456" height="890" alt="Image" src="https://github.com/user-attachments/assets/2b8e0bdd-fb23-419c-b385-b015a29e1949" />

### 必须的复现代码

无

### 错误信息

主动log的数据记录正常，系统标签页空白

**Comment by SAKURA-CAT** [comment]
有几个可能原因：
1. 你通过swanlab.Settings关闭了硬件监控功能
2. Slurm环境对硬件有一些特殊的虚拟化或者权限限制，swanlab无法通过一些库获取系统信息


你可以看看环境页面有没有相关信息，如果没有硬件信息则说明采集失败了。

<img width="892" height="317" alt="Image" src="https://github.com/user-attachments/assets/fd4d71b0-c0e2-4fa7-9fc3-e4b788d46671" />

此外，可以尝试设置`SWANLAB_LOG_LEVEL`环境变量为`debug`，看看有没有意外的错误

**Comment by LiaoLW** [comment]
> 有几个可能原因：
> 
> 1. 你通过swanlab.Settings关闭了硬件监控功能
> 2. Slurm环境对硬件有一些特殊的虚拟化或者权限限制，swanlab无法通过一些库获取系统信息
> 
> 你可以看看环境页面有没有相关信息，如果没有硬件信息则说明采集失败了。
> 
> <img alt="Image" width="892" height="317" src="https://private-user-images.githubusercontent.com/79990647/524682690-fd4d71b0-c0e2-4fa7-9fc3-e4b788d46671.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjUzNTY2MzAsIm5iZiI6MTc2NTM1NjMzMCwicGF0aCI6Ii83OTk5MDY0Ny81MjQ2ODI2OTAtZmQ0ZDcxYjAtYzBlMi00ZmE3LTlmYzMtZTRiNzg4ZDQ2NjcxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTEyMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMjEwVDA4NDUzMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTc0YmI1YmNmZDdhMmM4OGVlYTk2NWI5M2E3MGNmMDM2M2FhN2M3NGIzMmYxZDJkMzQzYjQ4OTI3OTQ1MGY0NjAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.KkYtocVWltZSmJZ3ZrsgdNZw4lNnutqrdXwTOfrwpEI">
> 此外，可以尝试设置`SWANLAB_LOG_LEVEL`环境变量为`debug`，看看有没有意外的错误

可以看到正确的GPU和CPU信息，所以可以推测不是上面的两个原因吗？

**Comment by SAKURA-CAT** [comment]
> 可以看到正确的GPU和CPU信息，所以可以推测不是上面的两个原因吗？

那尝试设置`SWANLAB_LOG_LEVEL`环境变量为`debug`，看看有没有意外的错误，看起来应该是上面的问题，而是出了一些意外错误

**Comment by LiaoLW** [comment]
> 有几个可能原因：
> 
> 1. 你通过swanlab.Settings关闭了硬件监控功能
> 2. Slurm环境对硬件有一些特殊的虚拟化或者权限限制，swanlab无法通过一些库获取系统信息
> 
> 你可以看看环境页面有没有相关信息，如果没有硬件信息则说明采集失败了。
> 
> <img alt="Image" width="892" height="317" src="https://private-user-images.githubusercontent.com/79990647/524682690-fd4d71b0-c0e2-4fa7-9fc3-e4b788d46671.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjUzNTg0NTQsIm5iZiI6MTc2NTM1ODE1NCwicGF0aCI6Ii83OTk5MDY0Ny81MjQ2ODI2OTAtZmQ0ZDcxYjAtYzBlMi00ZmE3LTlmYzMtZTRiNzg4ZDQ2NjcxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTEyMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMjEwVDA5MTU1NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQyMDVlZGI1ODNiZDRjZWJiNDNhNjlkYTFiMDUzYWE5ZDNmYzFhZTYxYmMwZmNmODQ0NDY1N2E0M2M1NDY4ZmMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.OjLu2bOLSJ5bggExhDmByXERkFygMllqrZcDpEq_5Ew">
> 此外，可以尝试设置`SWANLAB_LOG_LEVEL`环境变量为`debug`，看看有没有意外的错误

运行过程没有发现错误，log仅限于我自主上传的metric，没有看到硬件监控相关的信息：swanlab: MsgUploader is running
swanlab: MsgUploader is running
swanlab: MsgUploader is running
swanlab: HTTP Request: POST https://api.swanlab.cn/api/house/metrics | Response Status: 200 | Body: SUCCESS
swanlab: HTTP Request: POST https://api.swanlab.cn/api/house/metrics | Response Status: 200 | Body: SUCCESS
swanlab: MsgUploader is running
swanlab: Add data, key: train/loss.vis_loss_lvl6, step: 23009, data: 0.6712742745876312
swanlab: Add data, key: train/loss.loss, step: 23009, data: 10.465047597885132
swanlab: Add data, key: train/loss.geo_loss_lvl3, step: 23009, data: 0.762704998254776
swanlab: Add data, key: train/loss.geo_loss, step: 23009, data: 5.035974025726318
swanlab: Add data, key: train/loss.geo_loss_lvl1, step: 23009, data: 0.6907408237457275
swanlab: Add data, key: train/loss.geo_loss_lvl2, step: 23009, data: 0.7371421754360199
swanlab: Add data, key: train/loss.geo_loss_lvl5, step: 23009, data: 0.707690492272377
swanlab: Add data, key: train/loss.vis_loss_lvl3, step: 23009, data: 0.8542723953723907
swanlab: Add data, key: train/loss.vis_loss_lvl4, step: 23009, data: 0.7853100001811981
swanlab: Add data, key: train/loss.vis_loss_lvl5, step: 23009, data: 0.7619768381118774
swanlab: Add data, key: train/loss.geo_loss_lvl6, step: 23009, data: 0.623768150806427
swanlab: Add data, key: train/loss.vis_loss_lvl1, step: 23009, data: 0.8007548600435257
swanlab: Add data, key: train/loss.geo_loss_lvl0, step: 23009, data: 0.7800445109605789
swanlab: Add data, key: train/loss.vis_loss_lvl0, step: 23009, data: 0.8103812336921692
swanlab: Add data, key: train/loss.vis_loss_lvl2, step: 23009, data: 0.7451043128967285
swanlab: Add data, key: train/loss.vis_loss, step: 23009, data: 5.429073810577393
swanlab: Add data, key: train/loss.geo_loss_lvl4, step: 23009, data: 0.7338828295469284
swanlab: Add data, key: train/status.grad_norm, step: 23009, data: 3.9077205657958984
swanlab: Add data, key: train/learning_rate, step: 23009, data: 8.414341400903467e-05
swanlab: Add data, key: train/elastic.mem_ratio, step: 23009, data: 0.5
swanlab: Add data, key: train/elastic.params/k, step: 23009, data: 0.0009165540604556774
swanlab: Add data, key: train/elastic.memory, step: 23009, data: 17.463032960891724
swanlab: Add data, key: train/elastic.input_size, step: 23009, data: 15616.75
swanlab: Add data, key: train/elastic.params/b, step: 23009, data: 8.822614453368914
swanlab: Add data, key: train/grad_clip.max_norm, step: 23009, data: 1.0
swanlab: MsgUploader is running
swanlab: MsgUploader is running
swanlab: HTTP Request: POST https://api.swanlab.cn/api/house/metrics | Response Status: 200 | Body: SUCCESS
swanlab: HTTP Request: POST https://api.swanlab.cn/api/house/metrics | Response Status: 200 | Body: SUCCESS

**Comment by SAKURA-CAT** [comment]
> MsgUploader is running



> 运行过程没有发现错误，log仅限于我自主上传的metric，没有看到硬件监控相关的信息：swanlab: MsgUploader is running

不对，您需要打印试验一开始的指标，因为硬件监控会在实验一开始进行初始化，如果失败了会有相关的日志

**Comment by LiaoLW** [comment]
> > MsgUploader is running
> 
> > 运行过程没有发现错误，log仅限于我自主上传的metric，没有看到硬件监控相关的信息：swanlab: MsgUploader is running
> 
> 不对，您需要打印试验一开始的指标，因为硬件监控会在实验一开始进行初始化，如果失败了会有相关的日志

以下是初始化时的log：
⠙ Creating experiment...swanlab: HTTP Request: POST https://api.swanlab.cn/api/project | Response Status: 409 | Body: {'cuid': 'hmhhjwa1xz6mtyoh22a9j', 'name': 'Swanlab_debug', 'description': None, 
'visibility': 'PRIVATE', 'createdAt': '2025-12-06T07:01:23.369Z', 'updatedAt': '2025-12-12T13:09:57.259Z', 'path': 'liaolw/Swanlab_debug', 'role': 'OWNER', 'username': 'liaolw'}
⠹ Creating experiment...swanlab: HTTP Request: GET https://api.swanlab.cn/api/project/liaolw/Swanlab_debug | Response Status: 200 | Body: {'_count': {'experiments': 9, 'contributors': 1, 'children': 0, 
'collaborators': 0}, 'cuid': 'hmhhjwa1xz6mtyoh22a9j', 'name': 'Swanlab_debug', 'description': None, 'visibility': 'PRIVATE', 'createdAt': '2025-12-06T07:01:23.369Z', 'updatedAt': 
'2025-12-12T13:09:57.259Z', 'path': 'liaolw/Swanlab_debug', 'projectLabels': [], 'user': {'name': 'liaolw', 'avatar': 
'https://swanlab-beta-public-1301372061.cos.ap-nanjing.myqcloud.com/avatars/q4q7e6dth99h3hfwlj81ha6tqv2whua3', 'username': 'liaolw', 'status': 'ENABLED'}, 'profile': 
{'metadata': None, 'config': None, 'clusters': None}, 'lastRunTime': '2025-12-12T13:09:57.214Z', 'summary': {'cuid': 'k4em13w5uckde3y221he2', 'config': {'xAxis': {'key': 
'step', 'name': 'step', 'type': 'FLOAT', 'class': 'SYSTEM'}, 'smooth': {'type': 'TWE', 'value': 0.4}, 'showSmoothOriginal': True}}, 'role': 'CREATOR', 'permission': 'MANAGE', 
'group': {'username': 'liaolw', 'avatar': 'https://swanlab-beta-public-1301372061.cos.ap-nanjing.myqcloud.com/avatars/q4q7e6dth99h3hfwlj81ha6tqv2whua3', 'role': 'OWNER'}}
⠸ Creating experiment...swanlab: HTTP Request: POST https://api.swanlab.cn/api/project/liaolw/Swanlab_debug/experiment | Response Status: 201 | Body: {'cuid': 'fdizddt5qn4kaj5rjttng', 'type': 'CHAPTER', 
'name': 'swanlab_debug_1gpus', 'description': '2025-12-13 10:29:26', 'colors': ['#40877c', '#40877c'], 'show': True, 'pin': False, 'tags': [], 'state': 'RUNNING', 'config': None, 
'createdAt': '2025-12-13T02:29:28.347Z', 'finishedAt': None, 'user': {'username': 'liaolw', 'name': 'liaolw', 'avatar': 
'https://swanlab-beta-public-1301372061.cos.ap-nanjing.myqcloud.com/avatars/q4q7e6dth99h3hfwlj81ha6tqv2whua3', 'status': 'ENABLED'}, 'cloneType': 'ROOT', 'rootProId': None, 
'rootExpId': None, 'labels': [], 'sectionIndex': [], 'profile': {'config': None, 'metadata': None, 'requirements': None, 'conda': None}, 'flagId': 'ew0o7d'}
⠼ Creating experiment...swanlab: HTTP Request: GET https://api.swanlab.cn/api/project/liaolw/Trellis/runs/fdizddt5qn4kaj5rjttng/sts | Response Status: 200 | Body: {'pathStyle': False, 'expiredTime': 
1765600168, 'expiration': '2025-12-13T04:29:28Z', 'credentials': {'sessionToken': 
'***', 'tmpSecretId': 
'***', 'tmpSecretKey': '***'}, 'requestId': 
'01cbef6f-a51e-4269-8a00-eeadec1f442f', 'startTime': 1765592968, 'bucket': 'swanlab-beta-private-1301372061', 'region': 'ap-nanjing', 'prefix': 
'resources/hmhhjwa1xz6mtyoh22a9j/fdizddt5qn4kaj5rjttng'}
swanlab: The configuration is not saved because the setter is not set.
swanlab: MsgUploader is running
swanlab: SwanLab Runtime has initialized
swanlab: SwanLab will take over all the print information of the terminal from now on
swanlab: Tracking run with swanlab version 0.7.3
swanlab: Run data will be saved locally in ***
swanlab: 👋 Hi liaolw,welcome to swanlab!
swanlab: Syncing run swanlab_debug_1gpus to the cloud

**Comment by LiaoLW** [comment]
> > MsgUploader is running
> 
> > 运行过程没有发现错误，log仅限于我自主上传的metric，没有看到硬件监控相关的信息：swanlab: MsgUploader is running
> 
> 不对，您需要打印试验一开始的指标，因为硬件监控会在实验一开始进行初始化，如果失败了会有相关的日志

更新0.7.4后，新增两行报错：
swanlab: swanlab api error ('Trace id: None', 'PUT https://api.swanlab.cn/api/project/liaolw/Trellis_vae/runs/dw2e5hjt3iijpel8rk6pc/profile', '524 Receive timeout from 
origin')
swanlab: swanlab api error ('Trace id: None', 'POST https://api.swanlab.cn/api/experiment/dw2e5hjt3iijpel8rk6pc/columns', '524 Receive timeout from origin')

**Comment by SAKURA-CAT** [comment]
> > > MsgUploader is running
> > 
> > 
> > > 运行过程没有发现错误，log仅限于我自主上传的metric，没有看到硬件监控相关的信息：swanlab: MsgUploader is running
> > 
> > 
> > 不对，您需要打印试验一开始的指标，因为硬件监控会在实验一开始进行初始化，如果失败了会有相关的日志
> 
> 更新0.7.4后，新增两行报错： swanlab: swanlab api error ('Trace id: None', 'PUT https://api.swanlab.cn/api/project/liaolw/Trellis_vae/runs/dw2e5hjt3iijpel8rk6pc/profile', '524 Receive timeout from origin') swanlab: swanlab api error ('Trace id: None', 'POST https://api.swanlab.cn/api/experiment/dw2e5hjt3iijpel8rk6pc/columns', '524 Receive timeout from origin')

这是刚刚咱们云端服务有些波动，这会应该好了

**Comment by SAKURA-CAT** [comment]
> 以下是初始化时的log：

您应该没截取全，硬件监控正好在您这段日志后开始执行：

<img width="1057" height="155" alt="Image" src="https://github.com/user-attachments/assets/651a8fe5-de64-43d0-9a7e-f801f3afd2a7" />

那我们换个方式debug吧，您可以尝试在您的Slurm上执行如下内容：

```python

from swanlab.data.sdk import get_metadata


_, funcs = get_metadata()

print(funcs)

assert len(funcs) != 0, "硬件监控采集函数列表不能为空"
```

我不太清楚您的`len(funcs)`执行结果是什么，估计funcs是空列表

**Comment by SAKURA-CAT** [comment]
在 #1647 中实现了 `swanlab ping` 命令，可以尝试：
```
export SWANLAB_DEBUG=1
swanlab ping
```
然后贴出详细日志看看是哪里出问题了～
